sw/device/lib/dif/dif_pinmux.h

```diff
@@ -128,22 +147,31 @@ typedef uint32_t dif_pinmux_index_t;
  *   if such attribute is not specified. These include: `Drive*` and `SlewRate*`
  *   family of attributes.
  */
-typedef enum dif_pinmux_pad_attr {
-  kDifPinmuxPadAttrNone = 0,
-  kDifPinmuxPadAttrIOInvert = 1 << 0,
-  kDifPinmuxPadAttrVirtOpenDrain = 1 << 1,
-  kDifPinmuxPadAttrPullDown = 1 << 2,
-  kDifPinmuxPadAttrPullUp = 1 << 3,
-  kDifPinmuxPadAttrKeeper = 1 << 4,
-  kDifPinmuxPadAttrDriveWeakest = 1 << 5,
-  kDifPinmuxPadAttrDriveWeak = 1 << 6,
-  kDifPinmuxPadAttrDriveStrong = 1 << 7,
-  kDifPinmuxPadAttrDriveStrongest = 1 << 8,
-  kDifPinmuxPadAttrSchmittTrigger = 1 << 9,
-  kDifPinmuxPadAttrSlewRateSlow = 1 << 10,
-  kDifPinmuxPadAttrSlewRateFast = 1 << 11,
+typedef struct dif_pinmux_pad_attr {
+  dif_pinmux_pad_slew_rate_t slew_rate;
+  dif_pinmux_pad_drive_strength_t drive_strength;
+  bool invert_level : 1;
```

Comment:
:grimacing: I am not really comfortable with using bitfields here. We tend to avoid them, even though I never got around to actually banning them. They're... not a part of C I'm comfortable adjudicating.

I don't believe we lose that much from not making these be bitfields...

Reply:
Done.